sdk/core/core-paging/test/getPagedAsyncIterator.spec.ts

```diff
@@ -123,6 +123,33 @@ describe("getPagedAsyncIterator", function () {
     assert.deepEqual([].concat(...receivedPages), collection);
   });
 
+  describe("Iterator over object", function () {
+    interface collectionObject {
+      elements: number[];
+      next: number;
+    }
+
+    it("should return an iterator over an object that can extract elements", async function () {
+      const collection: collectionObject = {
+        elements: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+        next: 0,
+      };
+      const pagedResult: PagedResult<collectionObject, PageSettings, number, number> = {
+        firstPageLink: 0,
+        async getPage() {
+          return Promise.resolve({ page: collection });
+        },
+        toElements: (page) => page.elements,
+      };
+      const iterator = getPagedAsyncIterator(pagedResult);
+      const expected = [];
+      for await (const val of iterator) {
+        expected.push(val);
+      }
```

Comment:
Can't wait for [`Array.fromAsync](https://github.com/tc39/proposal-array-from-async) to be a thing

Reply:
Once the proposal goes through, we can make the change accordingly